hostinfo/fileperms.lua

```diff
@@ -0,0 +1,111 @@
+--[[
+Copyright 2015 Rackspace
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+]]--
+local table = require('table')
+local los = require('los')
+local async = require('async')
+local exists = require('fs').exists
+local stat = require('fs').stat
+local band = bit.band
+local fmt = require('string').format
+
+--[[ Check file permissions ]]--
+local HostInfo = require('./base').HostInfo
+local Info = HostInfo:extend()
+function Info:initialize()
+  HostInfo.initialize(self)
+end
+
+function Info:run(callback)
+  if los.type() ~= 'linux' then
+    self._error = 'Unsupported OS for file permissions'
+    callback()
+    return
+  end
+
+  local fileList = {
+  '/etc/grub.conf',
```

Comment:
Looks like your somehow lost indentation between this and the double end below around line 96.